File: practice-algos/longest_Substring_Without_Repeating_Characters/gpt_sol.js
function lengthOfLongestSubstring(s) {
    let left = 0, maxLength = 0;
    const charSet = new Set();

    for (let right = 0; right < s.length; right++) {
        while (charSet.has(s[right])) {
            charSet.delete(s[left]);
            left++;
        }
        charSet.add(s[right]);
        maxLength = Math.max(maxLength, right - left + 1);
    }

    return maxLength;
}

File: practice-algos/longest_Substring_Without_Repeating_Characters/gpt_sol.py
def length_of_longest_substring(s):
    char_set = set()
    left = maxLength = 0

    for right in range(len(s)):
        while s[right] in char_set:
            char_set.remove(s[left])
            left += 1
        char_set.add(s[right])
        maxLength = max(maxLength, right - left + 1)

    return maxLength
